ProcessLib/TimeLoop.cpp

```diff
@@ -550,16 +561,45 @@ static std::string const nonlinear_fixed_dt_fails_info =
     "Nonlinear solver fails. Because the time stepper FixedTimeStepping is "
     "used, the program has to be terminated.";
 
+void postTimestepForAllProcesses(
+    double const t, double const dt,
+    std::vector<std::unique_ptr<ProcessData>> const& per_process_data,
+    std::vector<GlobalVector*> const& _process_solutions,
+    std::vector<std::reference_wrapper<GlobalVector const>> const&
+        solutions_of_coupled_processes)
+{
+    // All _per_process_data share the first process.
+    bool const is_staggered_coupling =
+        !isMonolithicProcess(*per_process_data[0]);
+
+    for (auto& process_data : per_process_data)
+    {
+        auto const process_id = process_data->process_id;
+        auto& pcs = process_data->process;
+
+        if (is_staggered_coupling)
+        {
+            CoupledSolutionsForStaggeredScheme coupled_solutions(
+                solutions_of_coupled_processes, dt, process_id);
+            pcs.setCoupledSolutionsForStaggeredScheme(&coupled_solutions);
+        }
+        auto& x = *_process_solutions[process_id];
+        pcs.postTimestep(x, t, dt, process_id);
+        pcs.computeSecondaryVariable(t, x, process_id);
+    }
+}
+
 NumLib::NonlinearSolverStatus TimeLoop::solveUncoupledEquationSystems(
     const double t, const double dt, const std::size_t timestep_id)
 {
-    NumLib::NonlinearSolverStatus nonlinear_solver_status;
+    preTimestepForAllProcesses(t, dt, _per_process_data, _process_solutions);
+
+    NumLib::NonlinearSolverStatus nonlinear_solver_status{false, -1};
```

Comment:
For a readable source code, suggest to define the two variables for the arguments of the ctor of NumLib::NonlinearSolverStatus.  For example:

```suggestion
    const bool error_norms_met = false;
    const int number_iterations = -1
    NumLib::NonlinearSolverStatus nonlinear_solver_status{error_norms_met, number_iterations};
```